src/resources/packages/wizard/index.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useState } from "react";
+import domReady from "@wordpress/dom-ready";
+import ReactDOM from "react-dom";
+import { Modal } from "@wordpress/components";
+import { useEffect } from "@wordpress/element";
+import { useSelect, useDispatch } from "@wordpress/data";
+import OnboardingTabs from "./components/tabs";
+import { SETTINGS_STORE_KEY, MODAL_STORE_KEY } from "./data";
+
+import "./index.css";
+
+const OnboardingModal = ({ bootData }) => {
+    const [isInitialized, setIsInitialized] = useState(false); // Track initialization state.
+
+    // Dispatch actions
+    const { initializeSettings } = useDispatch(SETTINGS_STORE_KEY);
+    const { openModal } = useDispatch(MODAL_STORE_KEY);
+    const { closeModal } = useDispatch(MODAL_STORE_KEY);
+
+    // Initialize the settings store with boot data.
+    useEffect(() => {
```

Comment:
We shouldn't need the effects here as well. With appropriate dependency selecting of state we should be able to calculate during rendering